model/client.go

```diff
@@ -0,0 +1,201 @@
+package model
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// Client is the programmatic interface to the AWAT API.
+type Client struct {
+	address    string
+	headers    map[string]string
+	httpClient *http.Client
+}
+
+func NewClient(address string) *Client {
+	return &Client{
+		address:    address,
+		headers:    make(map[string]string),
+		httpClient: &http.Client{},
+	}
+}
+
+// CreateTranslation creates a new Translation which will start
+// shortly after being created
+func (c *Client) CreateTranslation(translationRequest *TranslationRequest) (*TranslationStatus, error) {
```

Comment:
I believe we should check for status codes in each clients method. From `http.Client` doc:
> A non-2xx status code doesn't cause an error.

As a safety net I guess it would also be good to defer closing of a response body.

Reply:
:+1: I was thinking about this a bit when looking at one of your recent PRs, where you used a bunch of `switch` statements for similar methods. I'll alter this to follow the same pattern.